src/components/BreedsList.vue

```diff
@@ -133,14 +126,47 @@ export default {
   computed: {
     displayedBreeds() {
       if (this.searchList.length > 0) {
+        this.inputIsInvalid = false;
         return this.searchList;
       } else if (this.searchList.length === 0 && this.searchQuery !== "") {
         this.searchList = [];
+        this.inputIsInvalid = true;
         return this.searchList;
       } else {
         return this.breedsList;
       }
     },
+    errorState() {
+      if (this.isLoading) {
+        return {
+          message: "The Data is Loading ...",
+        };
+      } else if (!this.isLoading && this.error) {
+        return {
+          message: this.error,
+        };
+      } else if (this.isTimeOut && !this.displayedBreeds.length) {
+        return {
+          message: "The request takes longer than expected...",
+        };
+      } else if (!this.isLoading && !this.breedsList.length) {
+        return {
+          message: "No Data Found. Please try again later.",
+        };
+      } else if (
+        !this.isLoading &&
+        this.searchQuery !== "" &&
+        !this.searchList.length
+      ) {
+        return {
+          message: ` Your searches for "${this.searchQuery}" did not have any matches.
+          Try different keywords.`,
+          class: "input__no-matches",
+        };
+      } else {
+        return false;
+      }
+    },
```

Comment:
Vse error message sem prestavil v computed property

Reply:
to lahko pustiva za nekej časa.. k si omenu da bi blo to dobr izboljšat sm mislu da bi izboljšal, ne samo prestavu v computed ;) 

če pogledaš data flow imaš par stepov:
1. a se data lovda a je že zlovdana
2. error ali success?

sepravi lahko recimo razdeliš v templateu na loading / ne loading (si dobu dato) in znotrej tega na error ali success.. 
in pol imaš samo 1 alpa mogoče 2 različna error msg, ki ju lahko določiš že ob fetchanju date..

s tem se izogneš 6 različnim `if...else` primeru kot je zgorej.. in je tud veliko bolj berljivo.. upam da sm razumljivo razložu

Reply:
> to lahko pustiva za nekej časa.. k si omenu da bi blo to dobr izboljšat sm mislu da bi izboljšal, ne samo prestavu v computed ;)
> 
> če pogledaš data flow imaš par stepov:
> 
> 1. a se data lovda a je že zlovdana
> 2. error ali success?
> 
> sepravi lahko recimo razdeliš v templateu na loading / ne loading (si dobu dato) in znotrej tega na error ali success.. in pol imaš samo 1 alpa mogoče 2 različna error msg, ki ju lahko določiš že ob fetchanju date..
> 
> s tem se izogneš 6 različnim `if...else` primeru kot je zgorej.. in je tud veliko bolj berljivo.. upam da sm razumljivo razložu

Se lahko tudi zdaj lotim tega ;) sem zgleda napačno zastopu pol oz sem si tako predstvljav

Reply:
@Luka85 lahko se tud sedaj lotiš.. up to you :D